sgkit/stats/pca.py

```diff
@@ -0,0 +1,338 @@
+from typing import Any, Optional, Union
+
+import dask.array as da
+import numpy as np
+import xarray as xr
+from dask_ml.decomposition import TruncatedSVD
+from sklearn.base import BaseEstimator
+from sklearn.pipeline import Pipeline
+from typing_extensions import Literal
+from xarray import DataArray, Dataset
+
+from ..typing import ArrayLike, DType, RandomStateType
+from ..utils import conditional_merge_datasets
+from .aggregation import count_call_alleles
+from .preprocessing import PattersonScaler
+
+
+def pca_est(
+    ds: Dataset,
+    n_components: int = 10,
+    *,
+    ploidy: Optional[int] = None,
+    scaler: Union[BaseEstimator, str] = "patterson",
```

Comment:
I tend to prefer having "None" as the default in the signature as it gives a bit more flexibility, both in the context of other parameters and over time as the API evolves. Are you sure that "patterson" will always be the right default value, in every context?

Same for other value here.

Reply:
Hm I can't think of a scenario where the default scaler becomes a choice made based on other inputs, but making it more flexible in case one comes up sounds good to me.  It definitely makes sense for the `algorithm` parameter.  I changed both in https://github.com/pystatgen/sgkit/pull/262/commits/28a9b495ffb38f92c526d0c91f62673f76d188bd.